### DISCUSSION TITLE
scraping of IPv6 link-local fe80 addresses fail

### DISCUSSION THREAD

**Discussion by emhamm** [prometheus/prometheus#12077] [Q&A]
### What did you do?

Versions tested with: 2.41.0, 2.37.5, 2.37.0
I am using file based service discovery to scrape targets in json format. 
One example file_sd_config:
```json
[ 
   {
        "labels": {
            "instance": "machine-test",
            "__metrics_path__": "/metrics-test"
        },
        "targets": [
            "[fe80::11:1111:c001:1111%bond0]:80"
        ]
    }
]
```

### What did you expect to see?

Successful Scrape of the target with the ip fe80::11:1111:c001:1111 through interface %bond0 on port 80

### What did you see instead? Under which circumstances?

```
Jan 19 11:22:42 machine-testing prometheus[1337]: ts=2023-01-19T10:22:42.751Z caller=scrape.go:1343 level=debug component="scrape manager" scrape_pool=testip6 target=http://[fe80::11:1111:c001:1111%25bond0]:80/metrics-mdi msg="Scrape failed" err="Get \\"http://[fe80::11:1111:c001:1111%25bond0]:80/metrics-test\": dial tcp [fe80::11:1111:c001:1111%bond0]:80: connect: invalid argument"
```

As one can see in this  log the address is correctly converted from an url to an uri and back (%zoneIdent => %25zoneIdent => %zoneIdent) so the zone identifier was correctly parsed, but still this error is thrown and nothing got scraped.

### System information

Linux 5.15.88 x64

### Prometheus version

```text
prometheus, version 2.41.0 (branch: unknown, revision: unknown)
  build user:       nix@nixpkgs
  build date:       unknown
  go version:       go1.19.4
  platform:         linux/amd64
```

### Prometheus configuration file

```yaml
no interessting things to see here
```

**Comment by dswarbrick** [comment]
Scoped link-local scrape targets are working for me (with interface "eno1"), with Prometheus 2.41.0, so I don't think your issue is caused by Prometheus.

Have you tried reproducing the error with a minimal Go program, performing a `http.Get("http://[fe80::11:1111:c001:1111%25bond0]:80/metrics-test")`? Since the "invalid argument" error is coming from the kernel, perhaps the issue lies with your IPv6 stack config.

Out of curiosity, what bonding mode is your interface bond0 using?
